Tutorials/drawing-options/drawing-options-functions-using-bitmap.md

```diff
@@ -0,0 +1,176 @@
+       return 0;
+    }
+    ```
+
+3. Function to Draw the Bitmap:
+
+   ```cpp
+   void draw_bitmap(bitmap bmp, double x, double y, drawing_options opts);
+   ```
+
+- `bmp`: The bitmap to draw.
+- `x`: The x-coordinate for the top-left corner of the bitmap.
+- `y`: The y-coordinate for the top-left corner of the bitmap.
+- `opts`: Drawing options to apply.
+
+## Drawing Options Functions
+
+## Example 1: Applying Option Part Bmp
+
+In this example, we'll demonstrate how to use 'Option Part Bmp' to draw a section of a bitmap, effectively cropping the image.
+
+1. **Function Syntax:**
+
+ ```cpp
+  drawing_options option_part_bmp(int x, int y, int width, int height);
+```
+
+- `x`: The x-coordinate for the top-left corner of the part of the bitmap to draw.
+- `y`: The y-coordinate for the top-left corner of the part of the bitmap to draw.
+- `width`: The width of the part of the bitmap to draw.
+- `height`: The height of the part of the bitmap to draw.
+
+1. **Example Usage:**
+
+   For instance, draw a window with dimensions of `800` pixels width and `600` pixels height. Here is the code snippet where we load a bitmap and applies 'Option Part Bmp' to draw only a 100x100 region of it at coordinates (100, 100).`bitmap bmp` loads a bitmap image named `my_bitmap` from the file `bitmap.png`. The bitmap is stored in the variable bmp for later use.
+
+   ```cpp
+   #include "spalshkit.h"
+   int main()
+    {
+       open_window("Drawing", 800, 600);
+
+       bitmap bmp = load_bitmap("my_bitmap", "bitmap.png");
+       drawing_options opts = option_part_bmp(0, 0, 100, 100);
+
+       draw_bitmap(bmp, 100, 100, opts);
+
+       delay(5000); // Pause for observation
+
+       close_all_windows();
+       return 0;
+    }
+
+## Example 2: Applying Option Rotate Bmp
+
+In this example, we'll showcase how to use 'Option Rotate Bmp' to dynamically rotate a bitmap at different angles.
+
+1. **Function Syntax:**
+
+ ```cpp
+  drawing_options option_rotate_bmp(double degrees);
+ ```
+
+- `degrees`: The angle of rotation in degrees.
+
+1. **Example Usage:**
+
+   In this code snippet, a window titled `Drawing` is initialized with dimensions of `800` pixels in width and `600` pixels in height using the open_window function. Next, a bitmap named `my_bitmap` is loaded from the file path `bitmap.png` using the load_bitmap function. Drawing options are applied to rotate the bitmap by `90 degrees clockwise` using the `option_rotate_bmp` function. Finally, the rotated bitmap is drawn onto the initialized window at coordinates (100, 100) using the `draw_bitmap` function.
+
+   ```cpp
+   #include "spalshkit.h"
+   int main()
+    {
+       open_window("Drawing", 800, 600);
+       
+       bitmap bmp = load_bitmap("my_bitmap", "bitmap.png");
+       drawing_options opts = option_rotate_bmp(90);
+
+       draw_bitmap(bmp, 100, 100, opts);
+
+       delay(5000);
+
+       close_all_windows();
+       return 0;
+    }
+
+## Example 3: Applying Option Scale Bmp
```

Comment:
This heading should be H3 (### instead of ##) as it is in the group of the heading above the first example section.